app/modules/conversations/conversation/conversation_service_copy.py

```
@@ -0,0 +1,804 @@
+
+    async def _update_conversation_title(self, conversation_id: str, new_title: str):
+        self.sql_db.query(Conversation).filter_by(id=conversation_id).update(
+            {"title": new_title, "updated_at": datetime.now(timezone.utc)}
+        )
+        self.sql_db.commit()
+
+    async def regenerate_last_message(
+        self,
+        conversation_id: str,
+        user_id: str,
+        node_ids: List[NodeContext] = [],
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)